File: docs/guides/dart/lib/savePopularRecords.dart
import 'package:algolia_client_search/algolia_client_search.dart';

void savePopularRecords() async {
  final client =
      SearchClient(appId: 'ALGOLIA_APPLICATION_ID', apiKey: 'ALGOLIA_API_KEY');

  final List<Map<String, dynamic>> records = [];

  BrowseResponse? browseResponse;
  do {
    final browseParams =
        BrowseParamsObject(hitsPerPage: 1000, cursor: browseResponse?.cursor);
    browseResponse =
        await client.browse(indexName: "indexName", browseParams: browseParams);
    for (final hit in browseResponse.hits) {
      final isPopular = hit['nbFollowers'] > 1000;
      final updatedProduct = {
        ...hit,
        'isPopular': isPopular,
      };
      records.add(updatedProduct);
    }
  } while (browseResponse.cursor != null);

  final batchParams = BatchWriteParams(
      requests: records
          .map((record) => BatchRequest(
                action: Action.addObject,
                body: record,
              ))
          .toList());
  await client.batch(
    indexName: "<YOUR_INDEX_NAME>",
    batchWriteParams: batchParams,
  );
}

File: docs/guides/csharp/src/savePopularRecords.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SavePopularRecords
{
  class Record
  {
    public int NbFollowers { get; set; }
    public required string TwitterHandle { get; set; }
    public bool IsPopular { get; set; }
  }

  async Task Main(string[] args)
  {
    var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));

    var hits = await client.BrowseObjectsAsync<Record>(
      "<YOUR_INDEX_NAME>",
      new BrowseParamsObject()
    );

    var records = hits.Select(hit => new Record
      {
        TwitterHandle = hit.TwitterHandle,
        NbFollowers = hit.NbFollowers,
        IsPopular = hit.NbFollowers >= 1_000_000,
      })
      .ToList();

    await client.SaveObjectsAsync("<YOUR_INDEX_NAME>", records);
  }
}
